src/repositories/indexable-hierarchy-repository.php

```diff
@@ -46,9 +67,37 @@ public function add_ancestor( $indexable_id, $ancestor_id, $depth ) {
 		return $hierarchy;
 	}
 
+	/**
+	 * Retrieves the ancestors. Create them when empty.
+	 *
+	 * @param Indexable $indexable The indexable to get the ancestors for.
+	 *
+	 * @return Indexable_Hierarchy[] The ancestors.
+	 */
+	public function get_ancestors( Indexable $indexable ) {
```

Comment:
Maybe this function should be renamed? I find it a bit misleading that a `get_` also sets (or builds in this case) something else.